### ISSUE TITLE
Allow to provide description to environment collection API

### ISSUE THREAD

**Issue by karrtikr** [microsoft/vscode#171108]
Particularly for the Python environments, the list of environment vars can be long, hence the prompt which shows up can be too verbose:

![image](https://user-images.githubusercontent.com/13199757/211768285-b4ceeae4-e99e-445a-85fe-cdafd65c3e08.png)

It would be perhaps more helpful if we can provide a simple description instead of the exact changes.

cc/ @Tyriar

**Comment by karrtikr** [comment]
This could also be extended to show up as `terminal.title` **the first time** the terminal is relaunched. Fish shell does not have a prompt variable which we could set like cmd to indicate which environment is active, which could also make use of this feature:

![image](https://user-images.githubusercontent.com/13199757/211916962-48fefb87-b8f2-4001-a887-14089a0d8652.png)

**Comment by Tyriar** [comment]
Proposal:

https://github.com/microsoft/vscode/issues/171108

```ts
export interface EnvironmentVariableCollection extends Iterable<[variable: string, mutator: EnvironmentVariableMutator]> {
	/**
	 * A description for the environment variable collection, this will be used to describe the
	 * changes in the UI.
	 */
	description: string | undefined;
}
```

We probably still want a way to show the actual environment variable changes even with this description though.

**Comment by meganrogge** [comment]
It would be nice if there were a `View Changes` action that opened a readonly editor with the changes. That would make it very accessible

**Comment by Tyriar** [comment]
Feedback from API sync: Add `MarkdownString`

**Comment by karrtikr** [comment]
### Updated proposal:

```typescript
export interface EnvironmentVariableCollection extends Iterable<[variable: string, mutator: EnvironmentVariableMutator]> {
	/**
	 * A description for the environment variable collection, this will be used to describe the
	 * changes in the UI.
	 */
	description: string | MarkdownString | undefined;
}
```

**Comment by Tyriar** [comment]
> description: string | MarkdownString | undefined;

My only concern here is we're putting the description into a markdown list which a `MarkdownString` could mess up the UI. We might want to either modify the description such that it's indented correctly to show up in a list item (1) or only take the description before the first `\n\n` (2):

For the description:

```md
"Hello

- abc
- def

`blah`"
```

1:

```md
"Hello
  
  - abc
  - def
  
  `blah`"
```

2:

```md
"Hello"
```

**Comment by karrtikr** [comment]
+1 for the second option to start with, as we expect the description to be concise. Personally I would only use Markdown to support backticks to represent commands.

**Comment by karrtikr** [comment]
Just realized description would be different for each scope, so proposal needs to be updated:

```typescript
export interface EnvironmentVariableCollection extends Iterable<[variable: string, mutator: EnvironmentVariableMutator]> {
    /**
     * Sets a description for the environment variable collection, this will be used to describe the changes in the UI.
     * @param description A description for the environment variable collection.
     * @param scope The scope to which this description applies to. If unspecified, operation applies to all
     * workspace folders.
     */
    setDescription(description: string | MarkdownString | undefined, scope?: EnvironmentVariableScope): void;
}

export type EnvironmentVariableScope = {
    /**
     * The workspace folder to which this collection applies to. If unspecified, collection applies to all workspace folders.
     */
    workspaceFolder?: WorkspaceFolder;
};
```

Blocked on https://github.com/microsoft/vscode/issues/171173

**Comment by karrtikr** [comment]
Is there a way to open a readonly editor rather than a untitled buffer?
![image](https://user-images.githubusercontent.com/13199757/233804202-8d792a13-e93f-4753-9ab2-94a1efb1463d.png)

**Comment by jrieken** [comment]
@Tyriar @karrtikr Looks like https://github.com/microsoft/vscode/issues/179476, https://github.com/microsoft/vscode/issues/171173, and https://github.com/microsoft/vscode/issues/171108 are all very closely related. For a holistic view at the changes a single discussion/issue/proposal might be more beneficial.

**Comment by Tyriar** [comment]
@jrieken created: https://github.com/microsoft/vscode/issues/182069

**Comment by karrtikr** [comment]
Closing in favor of https://github.com/microsoft/vscode/issues/182069 as these issues were only open to finalize proposed API.

**Comment by Tyriar** [comment]
@karrtikr they're all separate proposals, let's keep them open and have https://github.com/microsoft/vscode/issues/182069 for a wholistic view

**Comment by Tyriar** [comment]
I'm planning on finalizing this one independently of the workspace scope one this iteration.

**Comment by Tyriar** [comment]
To verify make sure the API shows up in the stable API and does not have a proposed check.

**Comment by DonJayamanne** [comment]
Tried this, not sure what I need to do to get this to work
I just updated the description property in the `context.environmentVariableCollection` and created a new terminal and nothing seems to be displayed.
@Tyriar Am i missing a step here?
